src/main/java/com/o19s/es/ltr/feature/store/index/IndexFeatureStore.java

```diff
@@ -53,44 +92,170 @@ public String getStoreName() {
     }
 
     @Override
-    public StoredFeature load(String id) throws IOException {
-        return getAndParse(FEATURE_TYPE, id, StoredFeature::parse);
+    public StoredFeature load(String name) throws IOException {
+        return getAndParse(name, StoredFeature.class, StoredFeature.TYPE);
     }
 
     @Override
-    public StoredFeatureSet loadSet(String id) throws IOException {
-        return getAndParse(id, FEATURE_SET_TYPE, StoredFeatureSet::parse);
+    public StoredFeatureSet loadSet(String name) throws IOException {
+        return getAndParse(name, StoredFeatureSet.class, StoredFeatureSet.TYPE);
+    }
+
+    /**
+     * Construct the elasticsearch index name based on a store name
+     */
+    public static String indexName(String storeName) {
+        if (Objects.requireNonNull(storeName).isEmpty()) {
```

Comment:
We throw a NPE with no message if `storeName` is `null` but an `IllegalArgumentException` if it is empty. This sounds like a job for Guava's `Preconditions.checkArgument()` and `Strings.isNullOrEmpty()`, but we don't have it as a dependency yet (probably not worth adding it, so feel free to ignore).